statwrap/fpp.py

```diff
@@ -585,3 +587,100 @@ def contingency_table(data, column_1, column_2):
 
     contingency_table = pd.crosstab(data[column_1], data[column_2])
     return contingency_table
+
+def expected_df(data, column_1, column_2, correction=False):
+    """
+    Generates expected frequency table.
+
+    Parameters
+    --------
+    data (pd.DataFrame): The original data frame.
+    column_1 (str): The name of the first column for the contingency table.
+    column_2 (str): The name of the second column for the contingency table.
+    correction (bool): Yates' correction for continuity, default is False.
+
+    Returns
+    --------
+    Displays expected frequency table
+
+    Examples
+    --------
+    >>> df = pd.read_csv("cps_categoricals_00.csv")
+    >>> expected_df(df, 'birth_continent-self', 'geo_region')
+    
+    (expected frequency table will appear in notebook output)
+    """
+    try:
+        # Create the observed frequency table
+        observed = contingency_table(data, column_1, column_2)
+        
+        # Perform the Chi-squared test
+        chi2, p, dof, expected = chi2_contingency(observed, correction=correction)
+
+        # Convert expected frequencies to DataFrame
+        expected_df = pd.DataFrame(expected, columns=observed.columns, index=observed.index).round(2)
```

Comment:
let's return `expected_df` and move the styling to be done in the chi2_test function